ecosystem/typescript/sdk/src/token_client.ts

```diff
@@ -46,26 +59,22 @@ export class TokenClient {
    * @returns A hash of transaction
    */
   async createCollection(account: AptosAccount, name: string, description: string, uri: string): Promise<string> {
-    const payload: Gen.TransactionPayload = {
-      type: "script_function_payload",
-      function: {
-        module: {
-          address: "0x3",
-          name: "token",
-        },
-        name: "create_collection_script",
-      },
-      type_arguments: [],
-      arguments: [
-        Buffer.from(name).toString("hex"),
-        Buffer.from(description).toString("hex"),
-        Buffer.from(uri).toString("hex"),
-        NUMBER_MAX.toString(),
-        [false, false, false],
-      ],
-    };
-    const transactionHash = await this.submitTransactionHelper(account, payload);
-    return transactionHash;
+    const payload = new TxnBuilderTypes.TransactionPayloadScriptFunction(
+      TxnBuilderTypes.ScriptFunction.natural(
+        "0x3::token",
+        "create_collection_script",
+        [],
+        [
+          BCS.bcsSerializeStr(name),
+          BCS.bcsSerializeStr(description),
+          BCS.bcsSerializeStr(uri),
+          BCS.bcsSerializeUint64(NUMBER_MAX),
```

Comment:
could we make the max a parameter in the createCollection with a default value of u64_max?